spec/update_pull_requests_milestone_spec.rb

```diff
@@ -0,0 +1,153 @@
+require 'spec_helper'
+
+describe Fastlane::Actions::UpdatePullRequestsMilestoneAction do
+  let(:test_token) { 'ghp_fake_token' }
+  let(:test_repo) { 'repo-test/project-test' }
+  let(:client) do
+    instance_double(
+      Octokit::Client,
+      user: instance_double('User', name: 'test'),
+      list_milestones: %w[12.2 12.3 12.4].map { |version| mock_milestone(version) },
+      'auto_paginate=': nil
+    )
+  end
+
+  def mock_milestone(version)
+    number = version.to_s.gsub('.', '').to_i
+    sawyer_resource_stub(title: "#{version} Fake milestone (##{number})", number: number)
+  end
+
+  def mock_pr(number)
+    sawyer_resource_stub(number: number)
+  end
+
+  before do
+    allow(Octokit::Client).to receive(:new).and_return(client)
+  end
+
+  context 'when providing a single PR number' do
+    it 'updates the milestone of a single PR' do
+      expect(client).to receive(:update_issue).with(test_repo, 1337, { milestone: 123 })
+
+      result = run_described_fastlane_action(
+        github_token: test_token,
+        repository: test_repo,
+        pr_number: 1337,
+        to_milestone: '12.3'
+      )
+
+      expect(result).to eq([1337])
+    end
+
+    it 'removes the milestone if to_milestone is nil' do
+      expect(client).to receive(:update_issue).with(test_repo, 1337, { milestone: nil })
+
+      result = run_described_fastlane_action(
+        github_token: test_token,
+        repository: test_repo,
+        pr_number: 1337,
+        to_milestone: nil
+      )
+
+      expect(result).to eq([1337])
+    end
+  end
+
+  context 'when providing a source milestone' do
+    it 'updates the milestone of all matching and still-opened PRs' do
+      allow(client).to receive(:search_issues)
+        .with(%(repo:#{test_repo} type:pr milestone:"#{mock_milestone(12.2)[:title]}" is:open))
+        .and_return({ items: [101, 103].map { |n| mock_pr(n) } })
+
+      expect(client).to receive(:update_issue).with(test_repo, 101, { milestone: 123 })
+      expect(client).to receive(:update_issue).with(test_repo, 103, { milestone: 123 })
+
+      result = run_described_fastlane_action(
+        github_token: test_token,
+        repository: test_repo,
+        from_milestone: '12.2',
+        to_milestone: '12.3'
+      )
+
+      expect(result).to eq([101, 103])
+    end
+
+    it 'adds a PR comment if one is provided' do
+      comment = 'Updated milestone from 12.2 to 12.3'
+      allow(client).to receive(:search_issues)
+                         .with(%(repo:#{test_repo} type:pr milestone:"#{mock_milestone(12.2)[:title]}" is:open))
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/wordpress-mobile/release-toolkit/blob/a45e842d8fb56f7fa70e4634b285dd8001a01adf/spec/update_pull_requests_milestone_spec.rb#L78"></span>Use 2 (not 19) spaces for indenting an expression spanning multiple lines.</td>
    </tr>
  </tbody>
</table>